### PR TITLE
Casmuser 3024 12 update raspberry upgrade docs to warn admins about uan can network outage

### PR DESCRIPTION
**Pull request by atifsyedali**
# Description

As per offline discussion about minimizing UAN disruption and a clear upgrade story for BICAN.

```
DOCS updates

1.  New CAN prep procedure - next NET person, possibly a no-op.
2.  Overview of UAN/UAI migration procedure to user segmented network. - Atif

   1.  Ahead of Raspberry upgrade

      1.  *[DONE]* Coordinate with casmnet for the application of the 1.2 preconfig for switches
      2.  *[DONE]* Identify subnet for new CAN
      3.  *[DONE]* Advise users that domain names will be changing, examples (in release notes already)

   2.  During Raspberry upgrade

      1.  At step 0.2

         1.  *[DONE]* Advise sysadmins that CFS will be temporarily disabled for UANs
         2.  *[DONE -- but need manual command to fix after reboot]* UAN reboots require rememdiation
         3.  *[DONE]* Access to UAIs will be disrupted until CSM upgrade completes and UAIs are restarted

   3.  After Raspberry upgrade

      1.  *[DONE: (pointer to https://github.com/Cray-HPE/docs-csm/pull/1854)]* Complete the migration of your users from CMN to new CAN
      2. *[DONE: (pointer to https://github.com/Cray-HPE/docs-csm/pull/1854)]* Run the operational procedure to remove CMN access to UANs.

3.  Warnings - Atif

   1.  *[NOT DONE: where is this getting started guide? -- don't think it is part of this repo]* Getting started guide
   2.  *[DONE]* CSM 1.2 upgrade readme
   3.  *[DONE]* Step 0.2 warning
```

# Checklist Before Merging

- [x] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [ ] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I don't have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/MTL-1695/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 1]
```diff
@@ -118,22 +119,178 @@ backup of Workload Manager configuration data and files is created. Once complet
...
+   
+   +- name: "Get {{ uan_user_access_cfg | upper }} CIDR from SLS, if network exists."
+   +  set_fact:
+   +    customer_access_network: "{{ item.CIDR }}"
+   +    customer_access_gateway: "{{ item.Gateway }}"
+   +  loop: "{{ sls_can.json[0].ExtraProperties.Subnets }}"
+   +  when: item.FullName == "CMN Bootstrap DHCP Subnet"
+   ```
+
+1. Once the CSM 1.2 upgrade is complete, you may reboot the UANs for the new network configuration changes to take effect.
```
[the remark this answers is not in the document]
Comment (atifsyedali) [reply upgrade/1.2/Stage_0_Prerequisites.md]: Changed thanks

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 2]
```diff
@@ -118,22 +119,178 @@ backup of Workload Manager configuration data and files is created. Once complet
...
+
+   > Note that the above command will disable CFS plays for UANs only. If you wish to disable CFS plays for all types of
+   > application nodes (recommended), then remove the `--subrole UAN` portion in the snippet above.
+
+1. UAN reboots must be avoided and is not a supported operation during CSM 1.2 upgrade. Rebooting a UAN during a CSM 1.2
+   upgrade can re-enable CFS and ultimately lead to removing the CMN interface from UANs, disrupting UAN access for your users.
+   As a system administrator, you must inform your users to avoid UAN reboots during the CSM 1.2 upgrade process.
+
+   If, however, a UAN is rebooted, then you need to patch the file `roles/uan_interfaces/tasks/can-v2.yml` for your current
+   CSM release in the `vcs/cray/uan-config-management.git` repository and re-run the CFS Ansible play to bring back the
```
[the remark this answers is not in the document]
Comment (atifsyedali) [reply upgrade/1.2/Stage_0_Prerequisites.md]: Changed as suggested. Thanks.

**Review by trad511 — APPROVED** [review]

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 3]
```diff
@@ -118,22 +119,178 @@ backup of Workload Manager configuration data and files is created. Once complet
...
+Prior to CSM 1.2, the single Customer Access Network (CAN) carried both the administrative network traffic and the user network
+traffic. CSM 1.2 introduces bifurcated CAN (BICAN), which is designed to separate administrative network traffic and user network traffic.
+With BICAN, the pre-1.2 CAN network is split into two separate networks:
+
+1. Customer Management Network (CMN)
+
+   This network allows only system administrative access from the customer site. The pre-1.2 CAN is renamed to CMN. By
+   the end of the CSM 1.2 upgrade, all non-administrative access, such as from UANs, will be removed from CMN.
+
+   During the CSM 1.2 upgrade, UANs will retain their pre-1.2 CAN IPs in order to minimize disruption to UANs. However,
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_0_Prerequisites.md:137]:
```suggestion
   During the CSM 1.2 upgrade, UANs will retain their pre-1.2 CAN IP addresses in order to minimize disruption to UANs. However,
```

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 4]
```diff
@@ -118,22 +119,178 @@ backup of Workload Manager configuration data and files is created. Once complet
...
+With BICAN, the pre-1.2 CAN network is split into two separate networks:
+
+1. Customer Management Network (CMN)
+
+   This network allows only system administrative access from the customer site. The pre-1.2 CAN is renamed to CMN. By
+   the end of the CSM 1.2 upgrade, all non-administrative access, such as from UANs, will be removed from CMN.
+
+   During the CSM 1.2 upgrade, UANs will retain their pre-1.2 CAN IP addresses in order to minimize disruption to UANs. However,
+   toward the end of the CSM 1.2 upgrade, UANs will stop registering themselves on CMN and will receive new IP addresses on the
+   CAN/CHN network below. This process is described in more detail in [UAN Migration](#uan-migration).
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_0_Prerequisites.md:139]:
```suggestion
   CAN/CHN network. This process is described in more detail in [UAN Migration](#uan-migration).
```

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 5]
```diff
@@ -118,22 +119,178 @@ backup of Workload Manager configuration data and files is created. Once complet
...
+
+   For user traffic only (e.g. users running and monitoring jobs), CSM 1.2 allows a choice of one of two networks:
+
+      - Customer Access Network (CAN) \[Recommended\]: this is a new network (VLAN6 in switches) that runs over the management network. This
+         network must not be confused with pre-1.2 CAN, which was a monolithic network that allowed both user and administrative
+         traffic, was configured as VLAN7 in switches, and is now renamed to CMN. The new CAN allows only user traffic.
+
+      - Customer High-speed Network (CHN) \[CSM 1.2 Tech Preview\]: this is a new network (VLAN5 in switches) that runs over the high-speed fabric.
+
+   You must only pick either the new CAN or CHN, but not both. Please note that CHN is a tech preview in CSM 1.2, and the new CAN is
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_0_Prerequisites.md:154]:
```suggestion
   Either the new CAN or CHN must be chosen, but not both. Note that the CHN is a technical preview in CSM 1.2, and the new CAN is
```

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 6]
```diff
@@ -118,22 +119,178 @@ backup of Workload Manager configuration data and files is created. Once complet
...
+   For user traffic only (e.g. users running and monitoring jobs), CSM 1.2 allows a choice of one of two networks:
+
+      - Customer Access Network (CAN) \[Recommended\]: this is a new network (VLAN6 in switches) that runs over the management network. This
+         network must not be confused with pre-1.2 CAN, which was a monolithic network that allowed both user and administrative
+         traffic, was configured as VLAN7 in switches, and is now renamed to CMN. The new CAN allows only user traffic.
+
+      - Customer High-speed Network (CHN) \[CSM 1.2 Tech Preview\]: this is a new network (VLAN5 in switches) that runs over the high-speed fabric.
+
+   Either the new CAN or CHN must be chosen, but not both. Note that the CHN is a technical preview in CSM 1.2, and the new CAN is
+   the recommended upgrade. The rest of the installation guide will provide you with options for configuring either the new CAN or CHN.
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_0_Prerequisites.md:155]:
```suggestion
   the recommended upgrade. The rest of the installation guide provides options for configuring either the new CAN or CHN.
```

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 7]
```diff
@@ -118,22 +119,178 @@ backup of Workload Manager configuration data and files is created. Once complet
...
+
+      - Customer High-speed Network (CHN) \[CSM 1.2 Tech Preview\]: this is a new network (VLAN5 in switches) that runs over the high-speed fabric.
+
+   Either the new CAN or CHN must be chosen, but not both. Note that the CHN is a technical preview in CSM 1.2, and the new CAN is
+   the recommended upgrade. The rest of the upgrade guide provides options for configuring either the new CAN or CHN.
+
+### UAN migration
+
+Certain steps are taken in order to minimize disruption to UANs during the CSM 1.2 upgrade process. Please read these steps
+carefully and follow any recommendations and warnings to minimize disruptions to user activity.  Note that these steps apply
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_0_Prerequisites.md:160]:
```suggestion
carefully and follow any recommendations and warnings to minimize disruptions to user activity. Note that these steps apply
```

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 8]
```diff
@@ -118,22 +119,178 @@ backup of Workload Manager configuration data and files is created. Once complet
...
+
+### UAN migration
+
+Steps are taken in order to minimize disruption to UANs during the CSM 1.2 upgrade process. Read these steps
+carefully and follow any recommendations and warnings to minimize disruptions to user activity. Note that these steps apply
+to all types of application nodes and not just UANs -- the term "UAN" just happens to be more commonly used and understood when
+referring to user activity.
+
+1. During the upgrade, the switch `1.2 Preconfig` will not remove UAN ports from the CMN VLAN (the pre-1.2 CAN), allowing UANs
+   to retain their existing IPs during the CSM 1.2 upgrade process. Traffic to and from UANs will still flow through CMN, but
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_0_Prerequisites.md:165]:
```suggestion
   to retain their existing IP addresses during the CSM 1.2 upgrade process. Traffic to and from UANs will still flow through CMN, but
```

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 9]
```diff
@@ -16,10 +16,11 @@ backup of Workload Manager configuration data and files is created. Once complet
...
 
 - [Stage 0.1 - Prepare assets](#prepare-assets)
-- [Stage 0.2 - Update SLS](#update-sls)
-- [Stage 0.3 - Upgrade Management Network](#update-management-network)
-- [Stage 0.4 - Prerequisites Check](#prerequisites-check)
-- [Stage 0.5 - Backup Workload Manager Data](#backup_workload_manager)
+- [Stage 0.2 - Plan and coordinate network upgrade](#plan-and-coordinate-network-upgrade)
+- [Stage 0.3 - Update SLS](#update-sls)
+- [Stage 0.4 - Upgrade Management Network](#update-management-network)
+- [Stage 0.5 - Prerequisites Check](#prerequisites-check)
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_0_Prerequisites.md:22]:
```suggestion
- [Stage 0.5 - Prerequisites check](#prerequisites-check)
```

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 10]
```diff
@@ -16,10 +16,11 @@ backup of Workload Manager configuration data and files is created. Once complet
...
 - [Stage 0.1 - Prepare assets](#prepare-assets)
-- [Stage 0.2 - Update SLS](#update-sls)
-- [Stage 0.3 - Upgrade Management Network](#update-management-network)
-- [Stage 0.4 - Prerequisites Check](#prerequisites-check)
-- [Stage 0.5 - Backup Workload Manager Data](#backup_workload_manager)
+- [Stage 0.2 - Plan and coordinate network upgrade](#plan-and-coordinate-network-upgrade)
+- [Stage 0.3 - Update SLS](#update-sls)
+- [Stage 0.4 - Upgrade Management Network](#update-management-network)
+- [Stage 0.5 - Prerequisites check](#prerequisites-check)
+- [Stage 0.6 - Backup Workload Manager Data](#backup_workload_manager)
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_0_Prerequisites.md:23]:
```suggestion
- [Stage 0.6 - Backup workload manager data](#backup_workload_manager)
```

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 11]
```diff
@@ -16,10 +16,11 @@ backup of Workload Manager configuration data and files is created. Once complet
 ### Stages
 
 - [Stage 0.1 - Prepare assets](#prepare-assets)
-- [Stage 0.2 - Update SLS](#update-sls)
-- [Stage 0.3 - Upgrade Management Network](#update-management-network)
-- [Stage 0.4 - Prerequisites Check](#prerequisites-check)
-- [Stage 0.5 - Backup Workload Manager Data](#backup_workload_manager)
+- [Stage 0.2 - Plan and coordinate network upgrade](#plan-and-coordinate-network-upgrade)
+- [Stage 0.3 - Update SLS](#update-sls)
+- [Stage 0.4 - Upgrade Management Network](#update-management-network)
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_0_Prerequisites.md:21]:
```suggestion
- [Stage 0.4 - Upgrade management network](#update-management-network)
```

**Review by mharding-hpe — APPROVED** [review]

**Pull request merged**